### PR TITLE
Sync file paths, metadata, and documentation with the problem-specifications repository

### PR DESCRIPTION
**Pull request by jagdish-15**
# Pull Request

This PR aims to synchronize all previously unsynced tests, file paths, metadata, and documentation with the **problem-specifications** repository.

### Proof of Sync
Screenshot 2024-11-19 102834
---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by jagdish-15** [comment]
As discussed in the [previous PR](https://github.com/exercism/java/pull/2867), I am planning to close that PR and have raised this one to focus on updating documentation, file paths, and metadata—essentially, all changes that do not require manual updates like tests.

For the test updates, I will raise seven separate PRs, one for each exercise, to ensure clarity and ease of review.

File: exercises/practice/dot-dsl/.meta/config.json [hunk 1]
```diff
@@ -16,6 +16,9 @@
     "editor": [
       "src/main/java/Node.java",
       "src/main/java/Edge.java"
+    ],
+    "invalidator": [
+      "build.gradle"
```
Comment (kahgoh) [inline exercises/practice/dot-dsl/.meta/config.json:21]: Did you mean to commit this change? It isn't clear to me why we would want to add this to the [invalidator](https://exercism.org/docs/building/tracks/practice-exercises#h-file-meta-config-json).

**Review by kahgoh — COMMENTED** [review]
Thanks @jagdish-15! I've just had a quick glance at the changes.

File: exercises/practice/dot-dsl/.meta/config.json (hunk 1 above)
Comment (jagdish-15) [reply exercises/practice/dot-dsl/.meta/config.json:21]: This was part of the `metadata` sync performed by `configlet` with the Problem Specifications repository.

**Comment by jagdish-15** [comment]
Hi @kahgoh,

This commit syncs the metadata of the *Pythagorean Triplet* exercise and the `instructions.md` for the *Complex Number* exercise, which I recently updated. The PR for the updated `instructions.md` file has been accepted in the problem-specifications repo.

I wanted to check if contributors typically add their names to the list of authors or contributors when updating an exercise's `instructions.md`. If not, that’s completely fine—I just want to ensure I’m following the usual practices.

Thanks!

File: exercises/practice/dot-dsl/.meta/config.json (hunk 1 above)
Comment (kahgoh) [reply exercises/practice/dot-dsl/.meta/config.json:21]: Ah, that makes sense! Thanks for letting me know.

**Comment by kahgoh** [comment]
> I wanted to check if contributors typically add their names to the list of authors or contributors when updating an exercise's `instructions.md`. If not, that’s completely fine—I just want to ensure I’m following the usual practices.

I think we normally _don't_ add to the contributors for syncing the `instructions.md`.

**Review by kahgoh — APPROVED** [review]

**Comment by jagdish-15** [comment]
> I think we normally _don't_ add to the contributors for syncing the `instructions.md`.

Hello @kahgoh, I believe there’s been a misunderstanding. I wasn’t referring to syncing the instructions.md. What I meant is that I authored the current introduction.md for the Complex Numbers exercise. My PR was recently accepted in the Problem Specifications repository. So, I wasn’t asking about the syncing process—I’m clarifying that I actually wrote the introduction.md in the Problem Specifications repo. I hope this clears up the confusion.

**Comment by kahgoh** [comment]
Hi @jagdish-15, looking at the past precedence (like, for example, in PR #2644, when the leap docs where re-synced after a rewrite) we still normally _don't_ add to the contributors. Adding the contributor (for changes to the Problem Specification) would require the tracks to be aware of the changes to know when to add to the contributor for the exercise across the tracks, which isn't always easy.

**Comment by jagdish-15** [comment]
Thank you so much for clarification!

**Pull request merged**